### DISCUSSION TITLE
Scripting in Template `.md` Files

### DISCUSSION THREAD

**Discussion by riyadh-h** [MarkBind/markbind#1649] [Q&A] [answered]
Hello there!

I am working on some modifications to the `default.md` template file. Some of these modifications involve scripting a few elements on the navbar. The script for these elements is written within `default.md` as an HTML `script` element as shown below.

![image](https://user-images.githubusercontent.com/69278589/128995734-1747eda1-d94c-4e62-8a41-5fcef05e92d7.png)

Judging from the original state of `default.md`, I assume that the scripting code should not be included within the template `.md` file. Therefore, I would like to know where it would be appropriate to write the script code within MarkBind's source files.

I am aware of the `vue-components` package, but I am not sure if it should include the files for elements that are not considered as MarkBind components (e.g., simple HTML buttons).

Thank you.
— Riyadh

**Comment by ang-zeyu** [comment, marked as answer]
Hi @riyadh-h! 

There should be no "appropriate" issue with this, we specially hoist any `<script> / <style>` tags in layout / pages to below the `<body>` tag automatically so it dosen't interfere with vue SSR in fact (which is generally unhappy with these tags inside templates as they introduce side effects). You could use `view page source` to double check for your case though, it might be a bug (if there's anything wrong).

There are no `<script>` tags in `default.md` simply because there's not much of a use case for them in a template site 😛

**Reply by riyadh-h** [reply]
Ah, I see.

Well, I tried writing the code for my changes (both scripts and HTML elements) in the form of a MarkBind plugin instead in order to maintain a more-organized template file, and they seem to work just as well.
I'll keep that in mind for future changes however 👍.

Thanks for the clarification, @ang-zeyu!
